src/client/Wrapper.js

```diff
@@ -73,6 +78,42 @@ export default class Wrapper extends React.PureComponent {
     this.props.getTrendingTopics();
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.locale !== this.props.locale) {
+      this.setState({ locale: this.getAvailableLocale(nextProps.locale) });
+      import(`react-intl/locale-data/${this.getAvailableLocale(nextProps.locale)}`).then((localeData) => {
+        addLocaleData(localeData);
+        import(`./locales/${this.getAvailableLocale(nextProps.locale)}.json`).then((data) => {
+          this.setState({ translations: data, loading: false });
+        });
+      });
+    }
+  }
+
+  getAvailableLocale = (appLocale) => {
+    let locale = appLocale || 'auto';
+
+    if (locale === 'auto') {
+      locale = this.getBrowserLocale() || 'en';
+    }
+
+    return locale;
+  };
+
+  getBrowserLocale = () => {
+    let detectedLocale;
+    if (typeof navigator !== 'undefined') {
+      detectedLocale =
+        navigator.userLanguage ||
```

Comment:
Lots of logic happening here, you need to split this up and break it down to make it more readable. You should not have multiple ternary operations like this, im surprised eslint didn't pick this up

Reply:
I don't think that's too bad here. To me it looks decent, except that `(navigator.languages && navigator.languages[0] ? navigator.languages[0] : undefined);` could be replaced just with `navigator.languages && navigator.languages[0]` and it does exactly the same thing.

eslint didn't complain, because there is are no nested ternary operators (which causes eslint errors).